pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -59,7 +87,411 @@ func (r *Builder) mapDisks(vm *model.Workload, persistentVolumeClaims []core.Per
+		Sockets: uint32(1),
+		Cores:   uint32(1),
+		Threads: uint32(1),
+	}
+}
+
+func (r *Builder) mapMemory(vm *model.Workload, object *cnv.VirtualMachineSpec) (err error) {
+	flavor := &model.Flavor{}
+	err = r.Source.Inventory.Find(flavor, ref.Ref{ID: vm.FlavorID})
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"Flavor lookup failed.",
+			"flavor",
+			vm.FlavorID)
+		return
+	}
+
+	reservation := resource.NewQuantity(int64(flavor.RAM*1024*1024), resource.BinarySI)
+	object.Template.Spec.Domain.Resources = cnv.ResourceRequirements{
+		Requests: map[core.ResourceName]resource.Quantity{
+			core.ResourceMemory: *reservation,
+		},
+	}
+
+	return nil
+}
+
+func (r *Builder) mapFirmware(vm *model.Workload, object *cnv.VirtualMachineSpec) {
+	features := &cnv.Features{}
+	firmware := &cnv.Firmware{}
+	firmware.Bootloader = &cnv.Bootloader{BIOS: &cnv.BIOS{}}
+	object.Template.Spec.Domain.Features = features
+	object.Template.Spec.Domain.Firmware = firmware
+}
+
+func (r *Builder) mapNetworks(vm *model.Workload, object *cnv.VirtualMachineSpec) (err error) {
+	var kNetworks []cnv.Network
+	var kInterfaces []cnv.Interface
+
+	kNetwork := cnv.Network{
+		Name:          "netos",
+		NetworkSource: cnv.DefaultPodNetwork().NetworkSource,
+	}
+	kInterface := cnv.Interface{
+		Name: "netos",
+	}
+	kInterface.Masquerade = &cnv.InterfaceMasquerade{}
+
+	kNetworks = append(kNetworks, kNetwork)
+	kInterfaces = append(kInterfaces, kInterface)
+	object.Template.Spec.Networks = kNetworks
+	object.Template.Spec.Domain.Devices.Interfaces = kInterfaces
+
+	return nil
+}
+
+// Build tasks.
+func (r *Builder) Tasks(vmRef ref.Ref) (list []*plan.Task, err error) {
+	vm := &model.Workload{}
+	err = r.Source.Inventory.Find(vm, vmRef)
+	if err != nil {
+		err = liberr.Wrap(
+			err,
+			"VM lookup failed.",
+			"vm",
+			vmRef.String())
+	}
+
+	for _, va := range vm.AttachedVolumes {
+		gb := int64(va.Size)
+		list = append(
+			list,
+			&plan.Task{
+				Name: fmt.Sprintf("%s-%s", r.Migration.Name, va.ID),
+				Progress: libitr.Progress{
+					Total: gb * 1024,
+				},
+				Annotations: map[string]string{
+					"unit": "MB",
+				},
+			})
+	}
+
+	return
+}
+
+// Create DataVolume certificate configmap.
+func (r *Builder) ConfigMap(_ ref.Ref, in *core.Secret, object *core.ConfigMap) (err error) {
+	return
+}
+
+func (r *Builder) DataVolumes(vmRef ref.Ref, secret *core.Secret, configMap *core.ConfigMap, dvTemplate *cdi.DataVolume) (dvs []cdi.DataVolume, err error) {
+	return nil, nil
+}
+
+// Build tasks.
+func (r *Builder) TemplateLabels(vmRef ref.Ref) (labels map[string]string, err error) {
+	return nil, nil
+}
+
+// Return a stable identifier for a DataVolume.
+func (r *Builder) ResolveDataVolumeIdentifier(dv *cdi.DataVolume) string {
+	return ""
+}
+
+// Return a stable identifier for a PersistentDataVolume
+func (r *Builder) ResolvePersistentVolumeClaimIdentifier(pvc *core.PersistentVolumeClaim) string {
+	return ""
+}
+
+// Build the DataVolume credential secret.
```

Comment:
ah that comment is still here, why do we still need this?

Reply:
Changed, but I think you previously commented in a different place because I copy-pasted it :)